src/app/fddev/tiles/fd_benchs.c

```diff
@@ -1,32 +1,337 @@
+/* _GNU_SOURCE for recvmmsg and sendmmsg */
+#define _GNU_SOURCE
+
 #include "../../fdctl/run/tiles/tiles.h"
+#include "../../../waltz/xdp/fd_xsk_aio.h"
+#include "../../../waltz/quic/fd_quic.h"
+#include "../../../waltz/tls/test_tls_helper.h"
 
 #include <linux/unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <string.h>
+#include <unistd.h>
+#include <poll.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <time.h>
+
+
+// TODO replace with config
+#define quic_enabled 1
+
+/* max number of buffers batched for receive */
+#define IO_VEC_CNT 128
+
+
+struct signer_ctx {
+  fd_sha512_t sha512[ 1 ];
+
+  uchar public_key[ 32UL ];
+  uchar private_key[ 32UL ];
+};
+typedef struct signer_ctx signer_ctx_t;
+
+
+static void
+signer( void *        _ctx,
+        uchar         signature[ static 64 ],
+        uchar const   payload[ static 130 ] ) {
+  fd_tls_test_sign_ctx_t * ctx = (fd_tls_test_sign_ctx_t *)_ctx;
+  fd_ed25519_sign( signature, payload, 130UL, ctx->public_key, ctx->private_key, ctx->sha512 );
+}
+
+static FD_FN_UNUSED
+signer_ctx_t
+signer_ctx( fd_rng_t * rng ) {
+  signer_ctx_t ctx[1];
+  FD_TEST( fd_sha512_join( fd_sha512_new( ctx->sha512 ) ) );
+  for( ulong b=0; b<32UL; b++ ) ctx->private_key[b] = fd_rng_uchar( rng );
+  fd_ed25519_public_from_private( ctx->public_key, ctx->private_key, ctx->sha512 );
+
+  return *ctx;
+}
+
+static int
+quic_tx_aio_send( void *                    _ctx,
+                  fd_aio_pkt_info_t const * batch,
+                  ulong                     batch_cnt,
+                  ulong *                   opt_batch_idx,
+                  int                       flush );
+
+
+/* quic_now is called by the QUIC engine to get the current timestamp in
+   UNIX time.  */
+
+static ulong
+quic_now( void * ctx ) {
+  (void)ctx;
+  return (ulong)fd_log_wallclock();
+}
 
 typedef struct {
   ulong round_robin_cnt;
   ulong round_robin_id;
 
   ulong packet_cnt;
 
-  ulong conn_cnt;
-  int conn_fd[ 128UL ];
+  ulong         conn_cnt;
+  int           conn_fd[ 128UL ];
+  struct pollfd poll_fd[ 128UL ];
+
+  signer_ctx_t     signer_ctx;
+  fd_quic_t *      quic;
+  ushort           quic_port;
+  fd_quic_conn_t * quic_conn;
+  const fd_aio_t * quic_rx_aio;
+  ulong            no_stream;
+
+  // vector receive members
+  struct mmsghdr rx_msgs[IO_VEC_CNT];
```

Comment:
Can we move all this QUIC stuff out to another file or like, `fd_quic_client_t` ? I feel like there could be a simple minimal interface which the bench tile can call without needing all of this stuff in its namespace.